test/ui/ValueElementTest.ts

```diff
@@ -0,0 +1,78 @@
+import { Facet, IFacetOptions } from '../../src/ui/Facet/Facet';
+import * as Mock from '../MockEnvironment';
+import { FacetValue } from '../../src/ui/Facet/FacetValues';
+import { ValueElement } from '../../src/ui/Facet/ValueElement';
+import { analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+import { ValueElementRenderer } from '../../src/ui/Facet/ValueElementRenderer';
+
+export function ValueElementTest() {
+
+  describe('ValueElementTest', () => {
+
+    let facet: Facet;
+    let facetValue: FacetValue;
+    let valueElementRenderer: ValueElementRenderer;
+    let valueElement: ValueElement;
+
+    beforeEach(() => {
+      facet = Mock.optionsComponentSetup<Facet, IFacetOptions>(Facet, {
+        field: '@field'
+      }).cmp;
+      facetValue = new FacetValue();
+      facetValue.value = 'value';
+
+      valueElementRenderer = new ValueElementRenderer(facet, facetValue);
+      spyOn(valueElementRenderer, 'setCssClassOnListValueElement');
+
+      valueElement = new ValueElement(facet, facetValue);
+      valueElement.renderer = valueElementRenderer;
+    });
+
+
+    it('should send one exclude UA event when excluding a facet value', () => {
+      facetValue.excluded = false;
+      facet.queryController.executeQuery = options => {
```

Comment:
The executeQuery method is mocked and that's how the UA events are sent. Other option is to assert the function that is sent to facet.triggerNewQuery which is not doable.